lib/views/main_screen.dart

```diff
@@ -1,52 +1,277 @@
+        /// `navNameIcon` Object maps Icons to proper features according to their names used in navbar.
+        /// CAUTION : Name of the feature in talwa app must match with the name that is provided by the admin.
+        final Map<String, Widget> navNameIcon = {
+          "Events": const Icon(Icons.event_note),
+          "Posts": const Icon(Icons.add_box),
+          "Profile": const Icon(Icons.account_circle),
+          "Chats": const Icon(Icons.chat_outlined),
+        };
 
+        /// `navNameClasses` Object maps the feature names with thier proper Icons and Widgets (named as `class`) used in navbar
+        /// CAUTION : Name of the feature in talwa app must match with the name that is provided by the admin.
+        final Map<dynamic, dynamic> navNameClasses = {
+          "Events": {
+            // "icon": Icons.event_note,
+            "class": ExploreEvents(
+              key: const Key('ExploreEvents'),
+              homeModel: model,
+            )
+          },
+          "Posts": {
+            "class": AddPost(
+              key: const Key('AddPost'),
+              drawerKey: MainScreenViewModel.scaffoldKey,
+            )
+          },
+          "Chats": {
+            "class": const ChatPage(
+              key: Key('Chats'),
+            ),
+          }
+        };
+
+        /// `navBarClasses` is the array that contains the Widgets to be rendered on the navbar
+        /// Features that should be implemented as plugins like Home, Profile,etc. should be kept here
+        /// When talawa app receives the plugins data is will dynamically render more components from the construtor.
+        navBarClasses = [
+          OrganizationFeed(key: const Key("HomeView"), homeModel: model),
+          ProfilePage(key: model.keySPEditProfile, homeModel: model),
+        ];
+        // ignore_for_file:  unused_local_variable
+        final res = renderBottomNavBarPlugins(
+            navBarItems, navBarClasses, navNameClasses, navNameIcon, context);
         return Scaffold(
           key: MainScreenViewModel.scaffoldKey,
-          drawer: CustomDrawer(
-            homeModel: model,
-            key: const Key("Custom Drawer"),
-          ),
+          drawer:
+              CustomDrawer(homeModel: model, key: const Key("Custom Drawer")),
           body: IndexedStack(
             index: model.currentPageIndex,
-            children: model.pages,
+            children: navBarClasses,
           ),
-          bottomNavigationBar: BottomNavigationBar(
-            type: BottomNavigationBarType.fixed,
-            currentIndex: model.currentPageIndex,
-            onTap: model.onTabTapped,
-            selectedItemColor: const Color(0xff34AD64),
-            items: model.navBarItems,
+          bottomNavigationBar: Stack(
+            children: [
+              SizedBox(
+                height: 50,
+                child: Row(
+                  children: [
+                    Expanded(
```

Comment:
This violates the DRY principle. Have a look at the refactored code and how it was converted into a function to avoid repetition.

Reply:
The current code is not compatible with the plugin logic.

Reply:
It is. Have a deeper thought over it. And if you still can't figure it out, then maybe change the logic a bit.